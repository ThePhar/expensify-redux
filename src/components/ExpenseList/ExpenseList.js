import React              from 'react';
import { connect }        from 'react-redux';
import getVisibleExpenses from '../../selectors/getVisibleExpenses';
import ExpenseListItem    from '../ExpenseListItem/ExpenseListItem';

export const ExpenseList = (props) => (
    <table>
        <tbody>
        {
            props.expenses.length === 0 ? (
                <tr><td>No expenses found.</td></tr>
            ) : (
                props.expenses.map((expense) => (<ExpenseListItem key={expense.id} {...expense} />))
            )
        }
        </tbody>
    </table>
);

const mapStateToProps = (state) => ({
    expenses: getVisibleExpenses(state.expenses, state.filters)
});

export default connect(mapStateToProps)(ExpenseList);