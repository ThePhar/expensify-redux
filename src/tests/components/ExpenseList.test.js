import { shallow }     from 'enzyme';
import React           from 'react';
import { ExpenseList } from '../../components/ExpenseList/ExpenseList';
import expenses        from '../fixtures/expenses';

test('Should render ExpenseList with expenses.', () => {
    const wrapper = shallow(<ExpenseList expenses={expenses} />);
    expect(wrapper).toMatchSnapshot();
});

test('Should render ExpenseList with empty message.', () => {
    const wrapper = shallow(<ExpenseList expenses={[]} />);
    expect(wrapper).toMatchSnapshot();
});