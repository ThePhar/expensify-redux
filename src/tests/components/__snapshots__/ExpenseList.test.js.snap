// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render ExpenseList with empty message. 1`] = `
<table>
  <tbody>
    <tr>
      <td>
        No expenses found.
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`Should render ExpenseList with expenses. 1`] = `
<table>
  <tbody>
    <ExpenseListItem
      amount={195}
      createdAt={0}
      description="Gum"
      id="1"
      key="1"
      note=""
    />
    <ExpenseListItem
      amount={109500}
      createdAt={-345600000}
      description="Rent bill"
      id="2"
      key="2"
      note=""
    />
    <ExpenseListItem
      amount={4500}
      createdAt={345600000}
      description="Credit Card"
      id="3"
      key="3"
      note=""
    />
  </tbody>
</table>
`;
